test/concatAll.js

```diff
@@ -0,0 +1,35 @@
+import { assert } from 'chai';
+import { NEL, Nil } from 'monet';
+
+import { concatAll } from '../src';
+import eq from './shared/eq';
+
+describe('concatAll', function() {
+  it('should concatenate arrays', function() {
+    eq(concatAll([[1], [2], [3]]), [1, 2, 3]);
+  });
+
+  it('should concatenate strings', function() {
+    eq(concatAll(['1', '2', '3']), '123');
+  });
+
+  it('should concatenate semigroups', function() {
+    eq(concatAll([NEL(1), NEL(2)]), NEL(1, NEL(2, Nil)));
+  });
+
+  it('should returns undefined if empty foldable was passed', function() {
+    eq(concatAll([]), undefined);
+  });
+
+  it('should throw if non-foldable is passed', function() {
+    assert.throws(() => concatAll(null), TypeError);
+  });
+
+  it('should throw if foldable contains non-semigroups', function() {
```

Comment:
Your test description contains branching logic. We use more proper mechanism for this.
```js
context('when foldable contains non-semigroups', function () {
  specify('should throw error', function () {
    assert.throws(() => concatAll([1, 2, null, true]), TypeError);
  });
});
```

Ref: https://github.com/char0n/ramda-adjunct/blob/a3109fc1020c6f4a8686d032cb7d539a230af258/test/notBoth.js#L41

Reply:
Done. It looks so neat now. Thanks, I have no experience in testing.